File: official/docs/csharp/v4/orders/buy.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using Newtonsoft.Json;
using EasyPost;
using EasyPost.Models.API;

namespace EasyPostExamples
{
    public class Examples
    {
        public static async Task Main()
        {
            string apiKey = Environment.GetEnvironmentVariable("EASYPOST_API_KEY")!;

            var client = new EasyPost.Client(apiKey);

            Order order = await client.Order.Retrieve("order_...");

            await order.Buy("FedEx", "FEDEX_GROUND");

            Console.WriteLine(JsonConvert.SerializeObject(order, Formatting.Indented));
        }
    }
}

File: official/docs/golang/v3/orders/buy.go
package example

import (
	"fmt"

	"github.com/EasyPost/easypost-go/v3"
)

func main() {
	client := easypost.New("EASYPOST_API_KEY")

	order, _ := client.BuyOrder("order_...", "FedEx", "FEDEX_GROUND")

	fmt.Println(order)
}
